csharp/Directory.Build.props

```diff
@@ -43,7 +43,7 @@
   <PropertyGroup>
     <Authors>The Apache Software Foundation</Authors>
     <PackageIconUrl>https://www.apache.org/images/feather.png</PackageIconUrl>
-    <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
```

Comment:
I'm definitely not a lawyer, but I am not sure this is appropriate.

The LICENSE.txt file in this repo has more information in it than the standard `Apache-2.0` license expression. I think we should leave the existing behavior in tact. Using a `PackageLicenseFile` is acceptable according to https://learn.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices#if-your-package-is-open-source.

Reply:
Oh, didn't see it has third-party notices. It mentions each file that has a separate license and none of them seems to be shipping with the C# NuGet packages.

Reply:
I think it would be safer to use the license file in the repo. It won't get out of date and it won't cause mistakes in the future.

Reply:
Reverted; we could place the third-party notices in a separate file and include that in the package (much like .NET does), but it's not urgent.